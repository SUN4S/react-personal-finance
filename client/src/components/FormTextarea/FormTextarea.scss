textarea {
  max-height: 150px;
  min-height: 70px;
  max-width: 100%;
  min-width: 100%;
  background-color: var(--page-bg);
  padding: 8px 0 0 6px;
}
