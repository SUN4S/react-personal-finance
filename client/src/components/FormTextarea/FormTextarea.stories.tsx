import { Meta, Story } from "@storybook/react/types-6-0";

import { FormTextarea } from "./FormTextarea";
import { FormTextareaProps } from "../../models/inputs";
import { useForm } from "react-hook-form";

interface TemplateProps extends FormTextareaProps {
  theme: string;
}

export default {
  title: "Components/Form Textarea",
  component: FormTextarea,
} as Meta;

const Template: Story<TemplateProps> = (args) => {
  const { register } = useForm();
  return (
    <html data-theme={args.theme}>
      <body
        style={{
          width: "400px",
          backgroundColor: "var(--sidebar-bg)",
          padding: "8px",
        }}
      >
        <FormTextarea
          labelFor={args.labelFor}
          label={args.label}
          name={args.name}
          required={args.required}
          placeholder={args.placeholder}
          register={register}
        />
      </body>
    </html>
  );
};

export const Default_Light = Template.bind({});
Default_Light.args = {
  theme: "light",
  labelFor: "name",
  label: "Enter Some text: ",
  name: "name",
  placeholder: "Text goes here",
  required: true,
};

export const Default_Dark = Template.bind({});
Default_Dark.args = {
  theme: "dark",
  labelFor: "name",
  label: "Enter Some text: ",
  name: "name",
  placeholder: "Text goes here",
  required: true,
};
