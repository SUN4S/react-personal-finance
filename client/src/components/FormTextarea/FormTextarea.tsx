import "./FormTextarea.scss";

import { FormTextareaProps } from "../../models/inputs";

// creates a form textarea input inside a label
export const FormTextarea: React.FC<FormTextareaProps> = ({
  labelFor,
  label,
  name,
  inputTestId,
  required,
  placeholder,
  register,
}) => {
  return (
    <label htmlFor={labelFor} data-testid="formLabel">
      {label}
      <textarea
        data-testid={inputTestId}
        {...register(name, { required: required })}
        placeholder={placeholder}
      />
    </label>
  );
};
