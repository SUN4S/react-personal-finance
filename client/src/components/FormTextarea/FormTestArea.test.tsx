import { cleanup, fireEvent, render } from "@testing-library/react";

import { FormTextarea } from "./FormTextarea";
import renderer from "react-test-renderer";

describe("Form Textarea", () => {
  afterEach(cleanup);
  const mockRegister = jest.fn();

  it("Renders", () => {
    const { getByTestId } = render(
      <FormTextarea
        labelFor="description"
        label="Description:"
        name="description"
        inputTestId="description"
        required={false}
        placeholder="Something to describe your expense"
        register={mockRegister}
      />
    );
    expect(getByTestId("formLabel")).toBeTruthy();
  });

  it("Render Label", () => {
    const { getByTestId } = render(
      <FormTextarea
        labelFor="description"
        label="Description:"
        name="description"
        inputTestId="description"
        required={false}
        placeholder="Something to describe your expense"
        register={mockRegister}
      />
    );
    expect(getByTestId("formLabel")).toHaveTextContent("Description:");
  });

  it("Renders Label For", () => {
    const { getByTestId } = render(
      <FormTextarea
        labelFor="description"
        label="Description:"
        name="description"
        inputTestId="description"
        required={false}
        placeholder="Something to describe your expense"
        register={mockRegister}
      />
    );
    expect(getByTestId("formLabel")).toHaveAttribute("for", "description");
  });

  it("Renders Placeholder", () => {
    const { getByTestId } = render(
      <FormTextarea
        labelFor="description"
        label="Description:"
        name="description"
        inputTestId="description"
        required={false}
        placeholder="Something to describe your expense"
        register={mockRegister}
      />
    );
    expect(getByTestId("description")).toHaveAttribute(
      "placeholder",
      "Something to describe your expense"
    );
  });

  it("Is Typeable", () => {
    const { getByTestId } = render(
      <FormTextarea
        labelFor="description"
        label="Description:"
        name="description"
        inputTestId="description"
        required={false}
        placeholder="Something to describe your expense"
        register={mockRegister}
      />
    );
    fireEvent.change(getByTestId("description"), {
      target: { value: "John" },
    });
    expect(getByTestId("description")).toHaveValue("John");
  });

  it("Matches Snapshot", () => {
    const tree = renderer.create(
      <FormTextarea
        labelFor="description"
        label="Description:"
        name="description"
        inputTestId="description"
        required={false}
        placeholder="Something to describe your expense"
        register={mockRegister}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
