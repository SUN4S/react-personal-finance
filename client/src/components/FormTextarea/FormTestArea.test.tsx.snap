// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Textarea Matches Snapshot 1`] = `
<label
  data-testid="formLabel"
  htmlFor="description"
>
  Description:
  <textarea
    data-testid="formTextarea"
    placeholder="Something to describe your expense"
  />
</label>
`;
